Sources/SystemInternals/Exports.swift

```diff
@@ -36,47 +50,129 @@ public var system_errno: CInt {
 #endif
 
 public func system_open(_ path: UnsafePointer<CChar>, _ oflag: Int32) -> CInt {
+#if os(Windows)
+  var fh: CInt = -1
+  _ = _sopen_s(&fh, path, oflag, _SH_DENYNO, _S_IREAD | _S_IWRITE)
+  return fh
+#else
   open(path, oflag)
+#endif
 }
 
+#if os(Windows)
+public func system_open(
+  _ path: UnsafePointer<CChar>, _ oflag: Int32, _ mode: CInt
+) -> CInt {
```

Comment:
I wonder if we should try to unify the interface around trivial type casts

Reply:
I was considering that myself, but wasn't sure if you would be okay with choosing the minimal type across the interfaces.

Reply:
If we want to do that, I think that doing a follow up might be the best way forward.